### PR TITLE
Make pipsi a package and make the inline scripts into files

### PR DESCRIPTION
**Pull request by pfmoore**

### CODE REVIEW

**Comment by pfmoore** [comment]
I'm not sure if the test failures here are problems with the PR. Some point to code I added, but others seem to be the "special characters" errors that appear to be hitting all PRs at the moment.

**Comment by pfmoore** [comment]
Rebased to pick up the test fixes in master

**Comment by pfmoore** [comment]
OK, looks like there are actual problems I need to fix...

**Comment by pfmoore** [comment]
Tests fixed, although the final fix just disables one of the "simple install" tests that uses `pipsi` as a package to install in a way that clashes with having a local directory called `pipsi`. I'm not particularly happy with doing that, but I can't see a better fix without some more significant surgery on how the test suite works (which I don't have the time to do right now).

File: testing/test_repo.py [hunk 1]
```diff
@@ -11,7 +11,7 @@ def repo(home, bin):
 
 @pytest.mark.parametrize('package, glob', [
     ('grin', 'grin*'),
-    ('pipsi', 'pipsi*'),
+#    ('pipsi', 'pipsi*'),
```
Comment (RonnyPfannschmidt) [inline testing/test_repo.py]: in order to keep track, please make this an `xfail` with a reason reference

Comment (pfmoore) [reply testing/test_repo.py]: OK, will do. It's a bit fiddly to just mark one set of parameters as xfail. I think I have the correct incantation, but if it fails again I'll work out how to fix it.

**Comment by RonnyPfannschmidt** [comment]
lets make an issue to track that

**Comment by pfmoore** [comment]
#115 created for this.

**Comment by RonnyPfannschmidt** [comment]
nicely done, thanks

**Pull request merged**